src/category_theory/limits/shapes/terminal.lean

```diff
@@ -61,6 +61,14 @@ is_limit.of_iso_limit hY
 { hom := { hom := i.hom },
   inv := { hom := i.inv } }
 
+def is_terminal.ite {Y Z : C} {P : Prop} [decidable P] (h : P) (hY : is_terminal Y) :
+  is_terminal $ if P then Y else Z :=
+hY.of_iso $ eq_to_iso $ by rw [if_pos h]
```

Comment:
Hmm I was overthinking it; `if_pos h ▸ hY` ought to work (if not, try `.symm ▸` or `.subst` or `.substr`); I guess we don't care about the data even though `is_terminal` isn't a Prop. Could it even be inlined?

Reply:
I have inlined `is_terminal.if_pos` and `is_terminal.if_neg` and deleted these two def/lemmas